service/src/main/java/uk/gov/hmcts/reform/fpl/service/JudicialService.java

```diff
@@ -549,4 +550,65 @@ private List<RoleAssignment> getAllocatedJudgeAndLegalAdvisorRoleAssignments(Lon
     private List<RoleAssignment> getHearingJudgeAndLegalAdviserRoleAssignments(Long caseId, ZonedDateTime endTime) {
         return roleAssignmentService.getCaseRolesAtTime(caseId, HEARING_ROLES, endTime);
     }
+
+    public List<JudicialUserProfile> getJudicialUserProfiles(JudicialUserRequest request) {
```

Comment:
I wonder if there's a way we could combine some of these so that we are maybe able to limit the number of requests to the judicial API? 

E.g. could we combine getJudgeTitleAndNameOfCurrentUser and isCurrentUserFeePaidJudge so we only have one call to get all this information?

Reply:
I have added Cacheable to JudicialApi.findUsers(). So it shouldn't call the judicial API multiple times.